corehq/motech/dhis2/static/dhis2/js/dhis2_entity_config.js

```diff
@@ -0,0 +1,77 @@
+hqDefine('dhis2/js/dhis2_entity_config', [
+    'jquery',
+    'knockout',
+    'hqwebapp/js/initial_page_data',
+    'hqwebapp/js/alert_user',
+    'hqwebapp/js/base_ace',
+], function (
+    $,
+    ko,
+    initialPageData,
+    alertUser,
+    baseAce
+) {
+    var caseConfig = function (data) {
+        var self = {};
+        self.oCaseConfig = ko.observable(JSON.stringify(data, null, 2));
+        self.toJSON = function () {
+            return JSON.parse(self.oCaseConfig());
+        };
+        return self;
+    };
+
+    var dhis2EntityConfig = function (caseConfigs) {
+        var self = {};
+        self.oCaseConfigs = ko.observableArray();
+
+        self.init = function () {
+            if (caseConfigs.length > 0) {
```

Comment:
Nitpick/style: this if block could be reduced to `self.oCaseConfigs().concat(_.map(caseConfigs, caseConfig));`

If only javascript believed arrays are falsy, the whole function could be one line: `self.oCaseConfigs.concat(_.map(caseConfigs || [{}], caseConfig))`

Reply:
:thinking: That looks cool, but using `.concat()` didn't seem to work for me.

Reply:
`self.oCaseConfigs().concat(...)` didn't do anything? or did it error?

Reply:
It doesn't error. I think it's changing the array returned by the observable, but it's not changing the value of the observable itself. ...

Never mind, I changed it to `self.oCaseConfigs(_.map(caseConfigs, caseConfig));` and that works.